src/preact.d.ts

```diff
@@ -45,7 +45,7 @@ declare namespace preact {
 
 		linkState:(name:string) => (event: Event) => void;
 
-		setState(state:StateType, opts?:any):void;
+		setState(state:StateType | ((prevState:StateType, props?:PropsType) => StateType), opts?:any):void;
```

Comment:
In a callback, don't make something optional. You are always allowed to give a callback with less parameters.